salt/modules/mac_service.py

```diff
@@ -114,205 +124,207 @@ def _service_by_name(name):
         if service['file_path'].lower() == name:
             # Match on full path
             return service
-        basename, ext = os.path.splitext(service['filename'])
+        basename, ext = os.path.splitext(service['file_name'])
         if basename.lower() == name:
             # Match on basename
             return service
 
-    return False
+    # Could not find service
+    raise CommandExecutionError('Service not found: {0}'.format(name))
 
 
-def start(service_path, domain='system'):
+def show(name):
```

Comment:
No, the service state only calls `start` with a single argument, the service name.  Also I have tested it doesn't work without the changes.